compiler/src/iree/compiler/Codegen/Dialect/LoweringConfig.h

```diff
@@ -98,11 +98,26 @@ inline void setTranslationInfo(
 // operations.
 // ===----------------------------------------------------------------------===//
 
+/// Returns the op that carries the `lowering_config` attribute; returns nullptr
+/// if none is carrying the attribute.
+///
+/// This scans ops in top-down order and the first one carrying the attribute
+/// will be returned.
+Operation *getLoweringConfigCarryingOp(ArrayRef<Operation *> computeOps);
```

Comment:
Could we make this `FailureOr<Operation *>`. Trying to avoid passing nullptrs around.

Reply:
Done. Though IMO it's pretty common across the MLIR codebases to use null `Operation *` or `Op` to represent single error state. `FailureOr` would be good if we'd like to differentiate different error states or covey the exact error reason. Seems an overkill here to use that here.

Reply:
That might be legacy (before we had `FailureOr` or `Optional`). I see MLIR moving towards using `FailureOr` in most places

Reply:
Also makes the ABI clearer. If you just pass a `nullptr` it isnt clear if the caller should account for `nullptr` or not. With the `FailureOr<..>` thats the indication to caller that the method might fail to get an operation. So the behavior of the function is clearer from the ABI.